app/scripts/lib/rpc-method-middleware/handlers/eth-accounts.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  JsonRpcRequest,
+  PendingJsonRpcResponse,
+  JsonRpcEngineNextCallback,
+  JsonRpcEngineEndCallback,
+} from 'json-rpc-engine';
+import { MESSAGE_TYPE } from '../../../../../shared/constants/app';
+
+type ethereumAccountsRequestType = {
+  methodNames: [string];
+  implementation: (
+    _req: JsonRpcRequest<unknown>,
+    res: PendingJsonRpcResponse<unknown>,
```

Comment:
- Type name should be capitalized.
- In the current version of `@metamask/json-rpc-engine`, `JsonRpcRequest` and `PendingJsonRpcResponse` are unable to accept `unknown` as generic arguments.
  - Even in the 3-years old version currently being imported in this file, `unknown` is not a good default for these generic params: these middleware types may need to be assignable to other downstream middleware, but `unknown` is not assignable to any type other than `unknown`.
- This type needs to expose the generic params of `JsonRpcRequest` and `PendingJsonRpcResponse` to the consumer. Otherwise there's no way for the user to type this correctly.
  - The generic defaults make this non-breaking in terms of the API it presents.
- This change is blocked by the extension upgrading to `@metamask/json-rpc-engine`.

```suggestion
type EthereumAccountsRequestType<
  Params extends JsonRpcParams = JsonRpcParams,
  Result extends Json = Json
> = {
  methodNames: [string];
  implementation: (
    _req: JsonRpcRequest<Params>,
    res: PendingJsonRpcResponse<Result>,
```